### DISCUSSION TITLE
Segmentation fault with DOSXYZnrc and source 21 jobs in parallel

### DISCUSSION THREAD

**Discussion by ms1497** [nrc-cnrc/EGSnrc#1106] [Questions]
When I submit jobs to run in parallel using DOSXYZ and Source 21 about a third of the jobs will never finish. The only error I am seeing is in the .eo files and is a segmentation fault. When I run the same job interactively with a smaller number of histories (100 k vs 100 M) I am unable to reproduce the same error. Does anyone have any idea about possible causes of this error?
<img width="354" alt="Capture" src="https://github.com/nrc-cnrc/EGSnrc/assets/163921700/cedcc5e2-5e43-4277-9e94-b90429f2defc">

**Comment by rtownson** [comment]
Try parallelizing on half as many cores but the same number of histories, to see if this uses less memory and avoids the error. And/or try increasing the voxel size.

Make sure that if you are using a material surrounding your DOSXYZnrc phantom, that it is NOT vacuum (fixing this may require editing the input file manually, not in the GUI).

**Reply by ms1497** [reply]
I've been able to try all of your suggestions but I am not seeing any improvement. I've set the surround to air and tried halving the number of voxels but the same jobs are always failing. Is there anything else I could try?

**Reply by rtownson** [reply]
Make sure to do cleanup before running parallel jobs again, by deleting any .lock files and _w*.egsdat files.

Does the job run with a different source? Try running the BEAMnrc simulation on its own to generate a phase-space, and see if that crashes or not. As a last resort, you could run the BEAMnrc sims in parallel to generate many phase-spaces, and then use those as the sources for your DOSXYZnrc jobs (this is how we used to do it before source 21).

**Comment by Ahsong9527** [comment]
If you use the file transferred from the host to the VM as the input,it may will casuse such problem. The solution is to copy the content of the file you transferred in the VM , create a new text and paste the previously copied content into it(It needs to be done in VM).
